### PR TITLE
optimize images

### PR DESCRIPTION
**Pull request by marcobiedermann**
Optimize images without losing any quality using [ImageOptim](https://imageoptim.com/)

image

I optimize images automatically, I highly recommend using a service like [ImgBot](https://github.com/apps/imgbot)

### CODE REVIEW

**Comment by lrusak** [comment]
Thanks for this. Do you have any stats about the total file size savings? Also can you post some side by side comparisons of files?

File: docs/resources/android.svg [hunk 1]
```diff
@@ -1,17 +1 @@
...
...54328 1.227 1.2139 1.227h.82033v2.5312c0 .63281.50933 1.1437 1.1344 1.1437.62509 0 1.1321-.50932 1.1421-1.1328v-2.5312h1.538v2.5312c0 .63511.50933 1.1444 1.1344 1.1444.63512 0 1.1437-.50932 1.1437-1.1499v-2.5312h.83191c.67139 0 1.2139-.54791 1.2139-1.2178v-7.4316l-10.209.006.03086.002zm7.4455-2.2611c-.23923 0-.43216-.1928-.43216-.43214 0-.23538.19294-.43063.43216-.43217.23923 0 .43216.19688.43216.43217 0 .23917-.19294.43214-.43216.43214m-4.6766 0c-.23923 0-.43216-.1928-.43216-.43214 0-.23684.19294-.43063.43216-.43063.23923 0 .43216.19688.43216.43987 0 .24297-.19449.43757-.43988.43757m4.8541-2.238.7941-1.4593c.04629-.0811.01079-.17442-.05866-.21992-.08027-.0347-.17287-.0117-.21916.0694l-.8103 1.4662c-.66985-.31253-1.4323-.48616-2.2303-.48616-.79795 0-1.5573.16596-2.2318.46534l-.79795-1.4686c-.04089-.0755-.13892-.10413-.21452-.0618-.07718.0347-.10494.13884-.06175.20826l.79487 1.447c-1.5666.80799-2.6238 2.346-2.6238 4.1132h10.285c0-1.7672-1.0557-3.3053-2.6207-4.1132m-9.2127 4.3002c-.63049 0-1.1421.51319-1.1421 1.1445v4.7785c0 .63281.51242 1.1421 1.1452 1.1421.62817 0 1.1398-.50932 1.1398-1.1421v-4.7769c0-.62897-.50933-1.1406-1.1344-1.1406" fill="#fff" stroke-width=".77172"/></g></svg>
```
Comment (lrusak) [inline docs/resources/android.svg]: I'm not sure about changes like this and what is actually being done as this is a vector image file.

Comment (marcobiedermann) [reply docs/resources/android.svg]: ImageOptim used SVGO under the hood to optimize SVGs. It optimizes all paths and removes some whitespace. It does not change the overall shape

image

See https://github.com/xbmc/xbmc/blob/ee8fd823e7afdf3bb3cd0d93c7c41cf87983cf16/docs/resources/android.svg

**Comment by marcobiedermann** [comment]
@lrusak 

As you can see the overall states are shown at the bottom of ImageOptim:
image

Unfortunately GitHub does not provide a side-by-side comparison right in the diffs.
Here is an example comparison between this PR and master

image
left: master, right: #16111

I am using ImageOptim since years and never had any quality issues.

On their website they claim:

> Removes bloated metadata. Saves disk space & bandwidth by compressing images without losing quality.

**Comment by stefansaraev** [comment]
there will be no saving at all after packing those skin images to xbt. this only makes sense where texturepacker is not used.

**Comment by marcobiedermann** [comment]
@stefansaraev 
You are right, it might not save a lot for the end consumer, still it is an optimization and saves time and bandwidth then working on this project overall.
It's also faster when it comes to processing those files.
Of course this impact might be rather small, still I see some improvements.

**Comment by candrews** [comment]
Should this be closed in favor of https://github.com/xbmc/xbmc/pull/16569 ?

**Pull request closed without merging**